src/components/toolbar/saved-notebooks-and-examples-subsection.jsx

```diff
@@ -1,47 +1,63 @@
 import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+
 import NotebookMenuItem from './notebook-menu-item'
 import NotebookMenuDivider from './notebook-menu-divider'
 import NotebookMenuHeader from './notebook-menu-header'
 import NotebookMenuSubsection from './notebook-menu-subsection'
 
+
 import iodideExampleTasks from '../../iodide-examples'
 import tasks, { getLocalStorageNotebook } from '../../task-definitions'
 import { stateFromJsmd } from '../../jsmd-tools'
 
-const AUTOSAVE = 'AUTOSAVE: '
-const autosave = Object.keys(localStorage).filter(n => n.includes(AUTOSAVE))[0] //
-const locallySaved = Object.keys(localStorage).filter(n => !n.includes(AUTOSAVE))
+class SavedNotebooksAndExamplesSubsection extends React.Component {
+  static propTypes = {
+    locallySaved: PropTypes.array,
+    autoSave: PropTypes.string,
+  }
 
-let autoSaveMenuItem
-let autoSave
-if (autosave !== undefined) {
-  autoSave = getLocalStorageNotebook(autosave)
-  autoSaveMenuItem = <NotebookMenuItem task={autoSave} key={autoSave.title} />
-}
+  getLocallySavedMenuItems() {
+    let locallySavedMenuItems
+    if (this.props.locallySaved) {
+      this.props.locallySaved.sort((a, b) => {
+        const p = (_) => {
+          let ls = localStorage.getItem(_)
+          if (!ls) return -1
+          ls = stateFromJsmd(ls)
+          return Date.parse(ls.lastSaved)
+        }
+        return p(b) - p(a)
+      })
 
-locallySaved.sort((a, b) => {
-  const p = (_) => {
-    let ls = localStorage.getItem(_)
-    if (!ls) return -1
-    ls = stateFromJsmd(ls)
-    return Date.parse(ls.lastSaved)
+      if (this.props.locallySaved.length) {
+        locallySavedMenuItems = this.props.locallySaved.map((l) => {
+          const task = getLocalStorageNotebook(l)
+          if (task !== undefined) {
+            return <NotebookMenuItem task={task} key={task.title} />
+          }
+          return undefined
+        })
+      }
+    }
+    return locallySavedMenuItems
   }
-  return p(b) - p(a)
-})
-
-let locallySavedMenuItems
-if (locallySaved.length) {
-  locallySavedMenuItems = locallySaved.map((l) => {
-    const task = getLocalStorageNotebook(l)
-    if (task !== undefined) {
-      return <NotebookMenuItem task={task} key={task.title} />
+
+  getAutoSaveMenuItem() {
```

Comment:
this function is probably not needed.

you can replace it and the block
```
        {autoSaveMenuItem ? <NotebookMenuHeader key="autosave" title="Auto-Saved" /> : undefined}
        {autoSaveMenuItem || undefined}
        {autoSaveMenuItem ? <NotebookMenuDivider key="autosave-divider" /> : undefined }
```

with something simpler in the render function like:

```
let autoSaveMenuItem
if (this.props.autoSave) {
autoSaveMenuItem = (<div> // FIXME: use React 16 fragments instead of useless container div
  <NotebookMenuHeader key="autosave" title="Auto-Saved" />  
  <NotebookMenuItem task={autoSave} key={autoSave.title} />
  <NotebookMenuDivider key="autosave-divider" />
</div> ) 
}
```

Now that i think of it, you can probalby do the same for getLocallySavedMenuItems()

(PS: i know that you are are just following the existing patterns, so not criticizing your PR -- i just didn't notice this stuff on the first code review, and now i'm seeing how we can clean this stuff up, so we may as well do it :-)  )